plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -391,9 +638,101 @@ func (c *CiscoTelemetryMDT) parseKeyField(tags map[string]string, field *telemet
 	}
 }
 
+func (c *CiscoTelemetryMDT) parseRib(grouper *metric.SeriesGrouper, field *telemetry.TelemetryField, prefix string, path string, tags map[string]string, timestamp time.Time) {
+        // RIB 
+        measurement := path
+        for _, subfield := range field.Fields {
+            //For Every table fill the keys which are vrfName, address and masklen
+            if subfield.Name == "vrfName" {
+                tags["vrfName"] = decodeTag(subfield)
+            }
+            if subfield.Name == "address" {
+                tags["address"] = decodeTag(subfield)
+            }
+            if subfield.Name == "maskLen" {
+                tags["maskLen"] = decodeTag(subfield)
+            }
+            if value := decodeValue(subfield); value != nil {
+                grouper.Add(measurement, tags, timestamp, subfield.Name, value)
+            }
+            if subfield.Name == "nextHop" {
+                //For next hop table fill the keys in the tag - which is address and vrfname
+                for _, subf := range subfield.Fields {
+                    for _, ff := range subf.Fields {
+                        if ff.Name == "address" {
+                            tags["nextHop/address"] = decodeTag(ff)
+                        }
+                        if ff.Name == "vrfName" {
+                            tags["nextHop/vrfName"] = decodeTag(ff)
+                        }
+                        if value := decodeValue(ff); value != nil {
+                            name := "nextHop/" + ff.Name
+                            grouper.Add(measurement, tags, timestamp, name, value)
+                        }
+                    }
+                }
+            }
+        }
+}
+ 
+
+func (c *CiscoTelemetryMDT) parseClassAttributeField(grouper *metric.SeriesGrouper, field *telemetry.TelemetryField, prefix string, path string, tags map[string]string, timestamp time.Time) {
+        // DME structure: https://developer.cisco.com/site/nxapi-dme-model-reference-api/
+        rn := ""
+        dn := false
+
+        var nxAttributes *telemetry.TelemetryField
+        nxAttributes = field
+        isDme := strings.Contains(path, "sys/")
+        if path == "rib" {
+           //handle native data path rib
+           c.parseRib(grouper, field, prefix, path, tags, timestamp)
+           return
+        }
+        if field == nil {
+            return
+        }
+        if !isDme {
+            return
+        }
+
+        //TODO: Need to protect it from causing panic
+        if field.Fields[0].Fields[0].Fields[0].Name != "attributes" {
+           return
+        }
+        nxAttributes = field.Fields[0].Fields[0].Fields[0].Fields[0]
+         
+        for _, subfield := range nxAttributes.Fields {
+                if subfield.Name == "rn" {
+                        rn = decodeTag(subfield)
+                } else if subfield.Name == "dn" {
+                        dn = true
+                        rn = decodeTag(subfield)
```

Comment:
You might want to `break` here. Furthermore, I do not understand the `"rn"` branch. Again it depends on the order in `nxAttributes.Fields` which value `rn` will have. Is this intentional? Anyway `rn` will never be used if not entering the `dn` part right?